src/main/java/cuchaz/enigma/bytecode/translators/TranslationMethodVisitor.java

```diff
@@ -119,58 +124,22 @@ public void visitTryCatchBlock(Label start, Label end, Label handler, String typ
 
 	@Override
 	public void visitLocalVariable(String name, String desc, String signature, Label start, Label end, int index) {
-		hasParameterMeta = true;
-
-		String translatedSignature = translator.translate(Signature.createTypedSignature(signature)).toString();
-		int argumentIndex = methodEntry.getArgumentIndex(ownerEntry, index);
-
-		if (argumentIndex >= 0) {
-			LocalVariableDefEntry entry = new LocalVariableDefEntry(methodEntry, index, name, true, new TypeDescriptor(desc));
-			LocalVariableDefEntry translatedEntry = translator.translate(entry);
-			String translatedName = translatedEntry.getName();
-
-			if (translatedName.equals(entry.getName())) {
-				List<TypeDescriptor> arguments = methodEntry.getDesc().getArgumentDescs();
-				List<TypeDescriptor> translatedArguments = arguments.stream()
-						.map(translator::translate)
-						.collect(Collectors.toList());
-
-				boolean argument = argumentIndex < arguments.size();
-				if (argument) {
-					translatedName = LocalNameGenerator.generateArgumentName(argumentIndex, translatedEntry.getDesc(), translatedArguments);
-				} else {
-					translatedName = LocalNameGenerator.generateLocalVariableName(argumentIndex, translatedEntry.getDesc());
-				}
-			}
+		signature = translator.translate(Signature.createTypedSignature(signature)).toString();
+		name = translateVariableName(index, name);
+		desc = translator.translate(new TypeDescriptor(desc)).toString();
 
-			super.visitLocalVariable(translatedName, translatedEntry.getDesc().toString(), translatedSignature, start, end, index);
-		} else {
-			// Handle "this" variable
-			TypeDescriptor translatedDesc = translator.translate(new TypeDescriptor(desc));
-			super.visitLocalVariable(name, translatedDesc.toString(), translatedSignature, start, end, index);
-		}
+		super.visitLocalVariable(name, desc, signature, start, end, index);
 	}
 
-	@Override
-	public void visitEnd() {
-		// If we didn't receive any parameter metadata, generate it
-		if (!hasParameterMeta) {
-			List<TypeDescriptor> arguments = translator.translate(methodEntry.getDesc()).getArgumentDescs();
-			int offset = ((methodEntry.getAccess().getFlags() & Opcodes.ACC_ABSTRACT) != 0) ? 1 : 0;
-
-			for (int argumentIndex = 0; argumentIndex < arguments.size(); argumentIndex++) {
-				LocalVariableEntry entry = new LocalVariableEntry(methodEntry, offset, "", true);
-				LocalVariableEntry translatedEntry = translator.translate(entry);
-				String translatedName = translatedEntry.getName();
-				if (translatedName.equals(entry.getName())) {
-					super.visitParameter(LocalNameGenerator.generateArgumentName(argumentIndex, arguments.get(argumentIndex), arguments), 0);
```

Comment:
Is the `LocalNameGenerator` no longer used?

Reply:
Oops, I'll add this back.

Reply:
Done.